build.gradle

```diff
@@ -20,17 +20,48 @@ apply plugin: MinorBumpPlugin
 apply plugin: VersionPrintPlugin
 apply plugin: ChangelogAddModelSectionPlugin
 
-
 buildscript {
     repositories {
+        // fetch plugins from google maven (https://maven.google.com)
+        google() {
+            content {
+                includeGroupByRegex("androidx\\..*")
```

Comment:
Why do we need to be so specific here?

Reply:
We can prevent inefficiencies and security risks of Gradle's default repository order. Gradle searches repositories linearly, slowing down builds and exposing the project to malicious artifacts. To optimize build times and enhance security, we can use repository-specific content filtering, including **GroupByRegex**, to limit searches to relevant repositories. This approach helps avoid unnecessary checks and ensures that only the required dependencies are fetched from the correct sources.

This article describes well about this approach: https://jasonatwood.io/archives/2033

Reply:
Due to improper ordering, Gradle couldn't download the JetBrains Compose compiler plugin from the Maven repository for some reasons, as shown in this attempt: https://github.com/GetStream/stream-chat-android/actions/runs/9589327132/job/26442900934